pkg/envoy/lds/multicluster_gateway.go

```diff
@@ -0,0 +1,83 @@
+package lds
+
+import (
+	"fmt"
+
+	xds_listener "github.com/envoyproxy/go-control-plane/envoy/config/listener/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"google.golang.org/protobuf/types/known/wrapperspb"
+
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/envoy"
+	"github.com/openservicemesh/osm/pkg/service"
+)
+
+func (lb *listenerBuilder) newMultiClusterGatewayListener() *xds_listener.Listener {
+	return &xds_listener.Listener{
+		Name:    multiclusterListenerName,
+		Address: envoy.GetAddress(constants.WildcardIPAddr, constants.EnvoyInboundListenerPort),
+		// TODO(steeling) for this to work on windows, there needs to be an inbound and an outbound listener
+		// see: https://www.envoyproxy.io/docs/envoy/latest/configuration/listeners/listener_filters/original_dst_filter#config-listener-filters-original-dst
+		// TrafficDirection: ...,
+		FilterChains: lb.getMultiClusterGatewayFilterChainPerUpstream(),
+		ListenerFilters: []*xds_listener.ListenerFilter{
+			{
+				// The OriginalDestination ListenerFilter is used to redirect traffic
+				// to its original destination.
+				Name: wellknown.OriginalDestination,
+			},
+		},
+	}
+}
+
+func (lb *listenerBuilder) getMultiClusterGatewayFilterChainPerUpstream() []*xds_listener.FilterChain {
+	var filterChains []*xds_listener.FilterChain
+
+	dstServices := lb.meshCatalog.ListMeshServicesForIdentity(lb.serviceIdentity)
```

Comment:
Will the gateway have a service identity ? as far as I am aware it won't, so you will not be able to get the mesh services from the gateway's identity